File: scala/play/book-api/app/entities/Book.scala
package entities

import play.api.libs.json.{Json, OFormat}

import java.time.LocalDateTime
import java.util.UUID

case class Book(bookId: UUID, title: String, synopsis: String, language: String, publisher: String, pages: Int, publicationDate: LocalDateTime)

object Book {
  implicit val bookFormat: OFormat[Book] = Json.format[Book]
}

File: go/gin/book-api/models/book.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Book struct {
	gorm.Model
	Id              uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey"`
	Title           string
	Synopsis        string
	Pages           int
	Language        string
	Publisher       string
	PublicationDate time.Time
}
